rails_programming/advanced_forms_and_activerecord/forms_advanced.md

```diff
@@ -193,3 +193,14 @@ This section contains helpful links to other content. It isn't required, so cons
 * [Using `inverse_of` to make `accepts_nested_attributes_for` work for `has_many :through` relationships](http://robots.thoughtbot.com/accepts-nested-attributes-for-with-has-many-through)
 * [Understanding Rails' form authenticity tokens](http://stackoverflow.com/questions/941594/understand-rails-authenticity-token)
 * [Why not to hardcode your application's secret token in production](http://daniel.fone.net.nz/blog/2013/05/20/a-better-way-to-manage-the-rails-secret-token/)
+
+### Knowledge Check
+This section contains questions for you to check your understanding of this lesson. If you’re having trouble answering the questions below on your own, review the material above to find the answer.
+
+- <a class="knowledge-check-link" href='#prepopulating-select-tags-with-collections'>What does the `#select_tag` helper do?</a>
+- <a class="knowledge-check-link" href='#select-knowledge-check'>When would you use the `#select` helper?</a>
+- <a class="knowledge-check-link" href='#options-knowledge-check'>When using `#options_for_select`, what format does the array need to be in?</a>
+- <a class="knowledge-check-link" href='#nested-forms'>How can you prevent users from having to submit multiple forms?</a>
+- <a class="knowledge-check-link" href='#model-knowledge-check'>What do you add to the model that allows nested forms to create new objects?</a>
+- <a class="knowledge-check-link" href='https://www.createdbypete.com/2014/04/04/working-with-nested-forms-and-a-many-to-many-association-in-rails-4.html'>How do you whitelist the nested parameters in your controller?</a>
+ - <a class="knowledge-check-link" href='#miscellania-blank-submissions-that-mean-delete'>What happens if you leave a form field (e.g. a checkbox) blank (unchecked)?</a>
```

Comment:
I think this question could be improved with a slight reworking. Currently, it implies that there is an absolute that occurs when a field is left blank when in reality it varies.

Reply:
I see what you mean. I'll rework this one, thanks!